feat: update settings page; use tanstack table for constraints/variants

markphelps opened the pull request:
1. Cleans up settings/preferences UI
2. ~~Give user ablity to set a default 'view mode' for views with potentially many items like constraints/variants~~
3. Updates variants and constraints to use tanstack table with better styling

![CleanShot 2025-05-05 at 12 08 03@2x](https://github.com/user-attachments/assets/fbfa291b-822f-442d-bfd4-83580f17a73a)
![CleanShot 2025-05-05 at 12 08 15@2x](https://github.com/user-attachments/assets/9ddb37cb-314c-4468-ac6b-2f609137e280)

erka commented on ui/src/app/preferences/Preferences.tsx:
```diff
@@ -1,94 +1,197 @@
+  const debounceTimerRef = useRef<NodeJS.Timeout | null>(null);
+  const DEBOUNCE_DELAY = 1000; // 1 second
+
+  // Show toast when preferences are saved, with debounce
+  useEffect(() => {
+    if (lastSaved) {
+      // Clear any existing timer
+      if (debounceTimerRef.current) {
+        clearTimeout(debounceTimerRef.current);
+      }
+
+      // Set new timer
+      debounceTimerRef.current = setTimeout(() => {
+        setNotification('Preferences saved', {
+          description: 'Your preferences have been automatically saved.',
+          duration: 2000
+        });
+        dispatch(resetLastSaved());
+        debounceTimerRef.current = null;
+      }, DEBOUNCE_DELAY);
+    }
+
+    // Cleanup timer on unmount
+    return () => {
+      if (debounceTimerRef.current) {
+        clearTimeout(debounceTimerRef.current);
+      }
+    };
+  }, [lastSaved, dispatch, setNotification]);
```
Could you please help me to understand why is it needed? Currently if I load the main page and go to Preferences I see the notification without any action.

markphelps replied:
i fixed the initial load issue, the goal is to prevent the dialog from showing up if you save prefs really quickly, like toggling the button on/off very fast. i only want the. notification to show up once in a given time period but im not sure its working as intended

erka commented on ui/src/components/constraints/Constraints.tsx:
```diff
@@ -202,6 +206,114 @@ function ConstraintCard({
+                )}
+              </td>
+              <td className="px-3 py-4 whitespace-nowrap text-right text-sm font-medium">
+                <button
```
```suggestion
                <button
                  type="button"
```

erka replied:
there is too more place in card view probably.

erka commented on ui/src/components/variants/Variants.tsx:
```diff
@@ -95,6 +99,106 @@ function VariantCard({
+                  )}
+                </td>
+                <td className="px-3 py-4 whitespace-nowrap text-right text-sm font-medium">
+                  <button
```
```suggestion
        <button
          type="button"

```

markphelps commented:
@erka i made the cards more compact as well as align the information better. I think its easier to read now?

> it's impossible to get the details using keyboard.

im not sure what you mean by this?

markphelps commented:
@erka I switched back to using only tables, and created new table components for variants and constraints.

I also updated them to use the same tanstack table library we use for namespaces table for performance and consistency and tried to update all styles to be similar

if you hover over the row an 'Edit' label will appear

i also fixed the keyboard navigation issue using 'tab' and 'enter' 

![CleanShot 2025-05-05 at 20 07 47@2x](https://github.com/user-attachments/assets/30ab5bbd-bbc4-4903-8f74-11a3368a08d1)
![CleanShot 2025-05-05 at 20 09 11@2x](https://github.com/user-attachments/assets/fb444385-b9cc-4a30-b9d6-b656a5c1be45)

erka reviewed (approved):
nice
